### PR TITLE
WHL scale testing - Log generation infra

### PR DESCRIPTION
**Pull request by chris-raitano**
## Summary

This PR sets up shared resources for building and deploying log generation containers for scale testing. This PR includes placeholders for Event Logs, ETWs, Crash Dumps and Text Logs. Additional PRs will be needed to implement the containers.

### Type of change:

- [ ] Feature work
- [ ] Bugfix
- [ ] Refactoring
- [ ] Documentation
- [ ] Developer Experience
- [x] Test
- [ ] Logging/Metrics
- [x] Automation 

## Developer Notes

Additional PRs should:

- Add a docker container to generate logs and a dockerfile to build the container
- Update deploy-log-generators.ps1 to build and deploy the new image (there are placeholders there)
- Update whl-scale-test.md with any configuration info (placeholders in section 5)
- Add any configuration options to log-generation-config.yaml

## Testing

- [ ] Unit tests
- [ ] Integration tests
- [x] Manual tests (add details)

This PR cannot be thoroughly tested on its own, but was tested as part of the separate log generation PRs.

### CODE REVIEW

File: test/whl-scale-tests/common.ps1 [hunk 1]
```diff
@@ -0,0 +1,170 @@
...
+windows server core image version to use
+.PARAMETER buildDirectory
+Root build directory. This should be where the dockerfile is located.
+.EXAMPLE
+Build-DockerImage exampleacr.azurecr.io/generatelogs:latest ltsc2022 . 
+#>
+function Build-DockerImage {
+    param(
+        [Parameter(Mandatory = $true)][string] $imageTag,
+        [Parameter(Mandatory = $true)][string] $windowsVersion,
```
Comment (cacarlt) [inline test/whl-scale-tests/common.ps1]: Current state of the scale infra: We only deploy Windows Server 2022 right now.

If we are going to give the users option here, then we should also provide this option in infra deployment. 

If we think scaling testing on both is going to be a big difference, then we should consider testing, but cost will double.

**Comment by cacarlt** [comment]
For every script that has to be ran independently we should explain in the .md file. Make sure to put it in the correct order with the rest of the scripts so they aren't run out of order.

File: test/whl-scale-tests/whl-scale-test.md
Comment (cacarlt) [inline test/whl-scale-tests/whl-scale-test.md:1]: For every script that has to be ran independently we should explain in the .md file. Make sure to put it in the correct order with the rest of the scripts so they aren't run out of order.

File: test/whl-scale-tests/common.ps1 [hunk 2]
```diff
@@ -0,0 +1,166 @@
+$ErrorActionPreference = 'Stop'
+$TempDir = "$PSScriptRoot\Temp"
+
+$WindowsVersion = "ltsc2022"
+$resourceGroupName = [Environment]::UserName + "scaletest"
+$acrName = $resourceGroupName + "acr"
+$aksClusterName = $resourceGroupName + "aks"
+$keyVaultName = $resourceGroupName + "kv"
+$genevaEnvironment = "DiagnosticsProd"
+$acrName = $resourceGroupName + "acr"
```
Comment (alrodrig1) [inline test/whl-scale-tests/common.ps1:9]: `$acrName` was already defined above.

File: test/whl-scale-tests/whl-scale-test.md [hunk 3]
```diff
@@ -0,0 +1,255 @@
...
+
+## Prerequisites
+- Access to an Azure Subscription
+- A Dignostic PROD Geneva Account where you have permissions to edit Geneva Log Configurations
+- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli)
+- Latest sky-dev branch
+<br>
+
+## 1. Deploy Scale Test Infrastructure 
+The first thing we need to do is setup the infrastructure for our scale test. We will do this with `.\Docker-Provider\test\whl-scale-tests\deploy-infra.ps1`. This script will create a new resource group (Ex. [alias]scaletest). Within that resource group you will find an AKS Cluster, an Azure Contianer Registry and a Key Vault. Your cluster will have four nodepool one for each scale component (Crash Dumps, Text Logs, ETW, Event Logs), by default, configured with the most recent stable kuberenetes version and each nodepool having one node each.
```
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: typo: Contianer

File: test/whl-scale-tests/whl-scale-test.md [hunk 4]
```diff
@@ -0,0 +1,255 @@
+# Windows Host Logs Scale Test
+
+## Prerequisites
+- Access to an Azure Subscription
+- A Dignostic PROD Geneva Account where you have permissions to edit Geneva Log Configurations
```
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: typo: Dignostic

File: test/whl-scale-tests/whl-scale-test.md (hunk 3 above)
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: typo: kuberenetes

File: test/whl-scale-tests/whl-scale-test.md [hunk 5]
```diff
@@ -0,0 +1,255 @@
...
+
+
+## 4. Taking Measurements of each component
+Need to actually deploy a test component something to get this filled out
+<br>
+
+## 5. How to clean up scale test infra
+To clean up all the resources you deployed we are going to use `.\Docker-Provider\test\whl-scale-tests\clean-infra.ps1`
+
+### Using clearn-infra.ps1
```
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: typo: clearn

File: test/whl-scale-tests/common.ps1 (hunk 1 above)
Comment (cacarlt) [reply test/whl-scale-tests/common.ps1]: This has been addressed

File: test/whl-scale-tests/whl-scale-test.md (hunk 3 above)
Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: Fixed

Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: fixed

File: test/whl-scale-tests/whl-scale-test.md (hunk 5 above)
Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: fixed

File: test/whl-scale-tests/whl-scale-test.md [hunk 6]
```diff
@@ -0,0 +1,255 @@
...
+-LocalPath "C:\C\7bb64a8ce0a1cff91c9489d224ed56dd78d46b9415a1acdafb670ebb72e665b1\opt\genevamonitoringagent\datadirectory" 
+-connectionInfo "AuthMSIToken" 
+"-serviceIdentity" "DiagnosticsProd#SkyMERProdLogs#SkyMERProdLogs_NAMESPACE#westus3" 
+"-configVersion" "6.0" 
+"-ManagedIdentity" 
+"client_id#10e472c6-ff9e-4656-a878-dc9644d40eab" 
+"-parentCpuRate" "25,6"
+-LogPath "C:\C\7bb64a8ce0a1cff91c9489d224ed56dd78d46b9415a1acdafb670ebb72e665b1\opt\genevamonitoringagent\datadirectory\Configuration\MonAgentHost.73.log"'
+   
+2. Check if the agent is having success with grabbing the config version
```
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: This step talks about checking for success, but the examples show errors instead. What should the user see when the agent successfully grabs the config?

File: test/whl-scale-tests/whl-scale-test.md [hunk 7]
```diff
@@ -0,0 +1,255 @@
...
+
+<br>
+
+## 2. Prepare Geneva XML Configuration
+
+> [!NOTE]
+>
+> If you know you already have a Geneva Account with each of these already configured go ahead and take note of their config versions because will need it for the next section. You will still need to setup your AKS Cluster Object ID for both Geneva Metrics and Geneva Logs
+
+In setting up the Windos Host Log Scale Test Suite you will need to make sure that you have four configurations that focus on each log type.
```
Comment (alrodrig1) [inline test/whl-scale-tests/whl-scale-test.md]: typo: Windos

File: test/whl-scale-tests/prepare-geneva-xml.ps1 [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+foreach ($filePath in $listOfFiles)
+{
+    Write-Host "Updating $filePath with your Geneva Account"
+
+    SubstituteNameValuePairs -InputFilePath $filePath -OutputFilePath $filePath -Substitutions $genevaXmlConfigurationHashTable
+
+    Write-Host "$filePath is ready to be uploaded to Geneva"
+}
+
+#Opens a new tab to the Geneva Metrics - Machine Access Section
```
Comment (alrodrig1) [inline test/whl-scale-tests/prepare-geneva-xml.ps1]: I think automatically opening a new tab might be perceived as a bit invasive. What if we just give the user the URL and next steps in the console?

Comment (cacarlt) [reply test/whl-scale-tests/prepare-geneva-xml.ps1]: How does "Print out the different filled in links for the user" sound?

File: test/whl-scale-tests/whl-scale-test.md
Comment (chris-raitano) [reply test/whl-scale-tests/whl-scale-test.md:1]: Added section 5 in the md

File: test/whl-scale-tests/whl-scale-test.md (hunk 4 above)
Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: Fixed

File: test/whl-scale-tests/whl-scale-test.md (hunk 6 above)
Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: Reworded to only go over failing/error cases

File: test/whl-scale-tests/whl-scale-test.md (hunk 7 above)
Comment (cacarlt) [reply test/whl-scale-tests/whl-scale-test.md]: Fixed

File: test/whl-scale-tests/prepare-geneva-xml.ps1 (hunk 8 above)
Comment (cacarlt) [reply test/whl-scale-tests/prepare-geneva-xml.ps1]: Fixed

**Comment by cacarlt** [comment]
Pease resolve conflicts and move this into a full PR state.

**Review by cacarlt — APPROVED** [review]
Great job @chris-raitano

**Pull request merged**